src/loaders/workspace_uploader/workspace_uploader.py

```diff
@@ -0,0 +1,557 @@
+    counter = 1
+    for assembly_name, assembly_dir in wait_to_upload_assemblies.items():
+
+        if counter % 5000 == 0:
+            print(f"{counter / assembly_files_len * 100:.2f}% jobs have been added to the queue {datetime.now()}")
+
+        assembly_path = os.path.join(JOB_DIR_IN_ASSEMBLYUTIL_CONTAINER, temp_dir, assembly_name)
+        conf.input_queue.put(
+            (
+                upload_env_key,
+                workspace_name,
+                workspace_id,
+                assembly_path,
+                assembly_dir,
+                assembly_name,
+                upload_dir,
+                counter,
+                assembly_files_len,
+            )
+        )
+        counter += 1
+
+    # Signal the workers to terminate when they finish uploading assembly files
+    for _ in range(conf.workers):
+        conf.input_queue.put(None)
+
+    results = [conf.output_queue.get() for _ in range(assembly_files_len)]
+    failed_names = [assembly_name for assembly_name, upa in results if upa is None]
+
+    # Close and join the processes
+    conf.pools.close()
+    conf.pools.join()
+
+    return failed_names
+
+
+def main():
+    parser = _get_parser()
+    args = parser.parse_args()
+
+    workspace_id = args.workspace_id
+    kbase_collection = getattr(args, loader_common_names.KBASE_COLLECTION_ARG_NAME)
+    source_version = getattr(args, loader_common_names.SOURCE_VER_ARG_NAME)
+    root_dir = getattr(args, loader_common_names.ROOT_DIR_ARG_NAME)
+    token_filepath = args.token_filepath
+    upload_file_ext = args.upload_file_ext
+    workers = args.workers
+    keep_job_dir = args.keep_job_dir
+
+    env = args.env
+    kb_base_url = loader_common_names.KB_BASE_URL_MAP[env]
+
+    if workspace_id <= 0:
+        parser.error(f"workspace_id needs to be > 0")
+    if workers < 1 or workers > cpu_count():
+        parser.error(f"minimum worker is 1 and maximum worker is {cpu_count()}")
+
+    uid = os.getuid()
+    username = os.getlogin()
+
+    job_dir = loader_helper.make_job_dir(root_dir, username)
+    collection_source_dir = loader_helper.make_collection_source_dir(
+        root_dir, loader_common_names.DEFAULT_ENV, kbase_collection, source_version
+    )
+    upload_dir = loader_helper.make_collection_source_dir(root_dir, env, kbase_collection, source_version)
+    output_dir = loader_helper.make_sourcedata_ws_dir(root_dir, env, workspace_id)
+
+    proc = None
+    conf = None
+
+    try:
+        # start podman service
+        proc = loader_helper.start_podman_service(uid)
+
+        # set up conf, start callback server, and upload assemblies to workspace
+        conf = Conf(
+            job_dir,
+            output_dir,
+            kb_base_url,
+            token_filepath,
+            loader_common_names.CALLBACK_UPLOADER_IMAGE_NAME,
+            workers,
+            _process_input,
+        )
+        workspace_name = conf.ws.get_workspace_info({"id": workspace_id})[1]
```

Comment:
I don't think you should need the workspace name anywhere, since AssemblyUtil will accept the ID.

Reply:
Only use workspace id now.

Reply:
:+1: